std/range/primitives.d

```diff
@@ -161,17 +161,11 @@ Params:
 Returns:
     true if R is an InputRange, false if not
  */
-template isInputRange(R)
-{
-    enum bool isInputRange = is(typeof(
-    (inout int = 0)
-    {
-        R r = R.init;     // can define a range object
-        if (r.empty) {}   // can test for empty
-        r.popFront;       // can invoke popFront()
-        auto h = r.front; // can get the front of the range
-    }));
-}
+enum bool isInputRange(R) =
+    is(typeof((R r) => R.init))
```

Comment:
My suggestion was not to delete the line entirely, but I don't see why it should be typeof((R r) => R.init) (with unused parameter r) instead of something like is(typeof(R.init) == R).